pokeapp/src/components/Sidebar.js

```diff
@@ -1,22 +1,67 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import PokeDollarIcon from "../assets/pokeDollar.svg";
 import {
   typeIconMapper,
   abilityMapper,
   gameMapper,
   weaknessIconMapper,
 } from "../utilities/mappers";
-import { useDispatch } from "react-redux";
+import { useDispatch, useSelector } from "react-redux";
 import { buyPokemon, sellPokemon } from "../redux/pokemonSlice";
+import { Typeahead } from "react-bootstrap-typeahead";
+import "react-bootstrap-typeahead/css/Typeahead.css";
+
+const sortAtoZ = (list) => {
+  const sortedAtoZ = list.sort((name1, name2) => {
+    if (name1 < name2) return -1;
+    if (name1 > name2) return 1;
+    return 0;
+  });
+  return sortedAtoZ;
+};
 
 const SideBar = ({
   selectedPokemon,
   makeUpperCase,
   isMoreInfo,
   setIsMoreInfo,
+  setSelectedPokemon,
 }) => {
+  const [showPokemon, setShowPokemon] = useState(false);
+  const [pokeNameList, setPokeNameList] = useState([]);
+
+  // Endpoint. This is the string that will be used by the type ahead
+  const [pokeSearch, setPokeSearch] = useState([]);
```

Comment:
Good job recognizing that this needs documentation, as its context is not immediately apparent.